src/citrine/resources/branch.py

```diff
@@ -100,38 +108,110 @@ def build(self, data: dict) -> Branch:
         return branch
 
-    def get_by_root_id(self, *, branch_root_id: Union[UUID, str]) -> Branch:
+    @deprecated(deprecated_in="2.42.0", removed_in="3.0.0", details="Use .get() instead.")
+    def get_by_root_id(self,
+                       *,
+                       branch_root_id: Union[UUID, str],
+                       version: Optional[Union[int, str]] = LATEST_VER) -> Branch:
         """
-        Given a branch root ID, retrieve the latest version of the branch.
+        Given a branch root ID and (optionally) the version, retrieve the specific branch version.
 
         Parameters
         ---------
-        branch_root_id:  Union[UUID, str]
+        branch_root_id: Union[UUID, str]
             Unique identifier of the branch root
 
+        version: Union[int, str], optional
+            The version of the branch to retrieve. If provided, must either be a positive integer,
+            or "latest". Defaults to "latest".
+
         Returns
         -------
         Branch
-            The latest version of the branch.
+            The requested version of the branch.
 
         """
-        params = {"root": str(branch_root_id), "version": "latest"}
-        branch = next(self._list_with_params(per_page=1, **params), None)
-        if branch:
-            return branch
+        return self.get(root_id=branch_root_id, version=version)
+
+    def get(self,
+            uid: Optional[Union[UUID, str]] = None,
+            *,
+            root_id: Optional[Union[UUID, str]] = None,
+            version: Optional[Union[int, str]] = LATEST_VER) -> Branch:
+        """
+        Retrieve a branch using either the version ID, or the root ID and version number.
+
+        Providing both the version ID and the root ID, or neither, will result in an error.
+
+        Providing the root ID and no version number will retrieve the latest version.
+
+        Using the version ID with this method is deprecated in favor of .get_by_version_id().
+
+        Parameters
+        ---------
+        uid: Union[UUID, str], optional
+            [deprecated] Unqiue ID of the branch version.
+
+        root_id: Union[UUID, str], optional
+            Unique identifier of the branch root
+
+        version: Union[int, str], optional
+            The version of the branch to retrieve. If provided, must either be a positive integer,
+            or "latest". Defaults to "latest".
+
+        Returns
+        -------
+        Branch
+            The requested version of the branch.
+
+        """
+        if uid:
+            warnings.warn("Retrieving a branch by its version ID using .get() is deprecated. "
+                          "Please use .get_by_version_id().",
+                          DeprecationWarning)
+
+            if root_id:
+                raise ValueError("Please provide exactly one: the version ID or root ID.")
+            return self.get_by_version_id(version_id=uid)
+        elif root_id:
+            version = version or LATEST_VER
```

Comment:
```suggestion
            if version is None:
                version = LATEST_VER
```
The original behaves incorrectly with empty string.

Reply:
No it doesn't. If the user provides an empty string, we should treat that the same as `None`.